rover.test.ts

```diff
@@ -0,0 +1,26 @@
+import Plateau from "./plateau"
+import Rover from "./rover"
+
+const explore = jest.requireActual('./explore')
+const mockedExplore = jest.spyOn(explore, 'default')
+
+describe('Rover', () => {
+    const config = {name: 'rover1', id: '0001'}
+    const plateau1 = new Plateau(3,5) 
+    const rover1 = new Rover(config)
+
+    // ignore
+    it('Should setup successfully', () => {
+        expect(rover1.setup()).toEqual('Setup successfull')
+    })
+
+    // ignore
```

Comment:
Ignore...as in?

Reply:
We have 2 dummy methods just for representation of rover class (as an example of what extra methods it could have). Hence, in test I've written test for them that don't serve any purpose but exist for demonstration purpose only (as an example).

Reply:
The rover class doesn't extend/implement any parent. Any reason to have the extra methods? Had it implemented an interface it would have made sense, but don't see the value here.

Reply:
Right, should have implemented interface I guess. The goal was to create rover that has access to methods inside class as in any rover could move, getlocation, takephoto for example, these methods are common for all rovers. Can't a single class itself act as an interface ? because all of its object are of type rover right.

Reply:
But do we have multiple rovers? I don't think the problem mentions different types of rovers.
Not sure what you mean by "class acting as an interface"

Reply:
We have to create multiple rovers right ? as in main.ts file we created 2 rovers and second rover will move once first one has completed moving.

Rover (class) -> rover1, rover2, rover3 (objects)

By interface I mean rover1, rover2, rover3 implements Rover interface. Of course we didn't explicitly create interface ( with key) but generally they all follow interface of Rover class right.

Reply:
Piyush, I'm talking about interfaces in OOPs, not typescript interfaces.

Reply:
I know you said in regards to OOPs. But I mean like all rovers we create are instance of Rover and has access to all methods. We just want all rovers to have a method called takePhoto(), move(), sleep() and so I've simply created dummy methods along with move.

Forget about interface statement, I caused confusion.